### DISCUSSION TITLE
Fixtures + setup methods behavior change between 6.x and 7.x

### DISCUSSION THREAD

**Discussion by amercader** [pytest-dev/pytest#10005] [Q&A]
We have a big legacy test suite that uses nose-style `setup()` methods on test classes and have noticed a behavior change between pytest 6.2.5 and pytest 7.0.1.

Consider the following test:

```python
# test_hello.py
@pytest.fixture
def hello():
    print("Hello")

@pytest.mark.usefixtures("hello")
class TestHello:

    # Same happens with setup_method(self, method)
    def setup(self):
        print("world")

    def test_hello(self):
        pass
```

In pytest 6.2.5, first the fixtures get executed, then the `setup()` methods:
```
pytest -s test_hello.py
======================================================== test session starts =========================================================
platform linux -- Python 3.8.10, pytest-6.2.5, py-1.10.0, pluggy-0.13.1
collected 1 item                                                                                                                     
test_hello.py Hello
world
.
```

In pytest 7.0.1, first the `setup()` method gets executed, then the fixtures:

```
pytest -s test_hello.py        
==================================================================== test session starts ====================================================================
platform linux -- Python 3.8.10, pytest-7.0.1, pluggy-0.13.1
collected 1 item                                                                                                                                            

test_hello.py world
Hello
.
```

Looking at the [Changelog](https://docs.pytest.org/en/7.0.x/changelog.html#) I couldn't find any mention of this change, is this a regression or it is indeed the new expected behavior?

Many thanks!

**Comment by NeroMorto** [comment]
We met the same behavior. @amercader Have you found any solution or explanation for this?
Furthermore, we found that `autouse=True` also changes the order of execution:
```python
# test_hello.py
import pytest

@pytest.fixture(autouse=True)
def hello():
    print("Hello")

class TestHello:

    def setup_method(self):
        print("world")

    # @pytest.mark.usefixtures("hello")
    def test_hello(self):
        pass
```
Result: 
```shell
pytest -s test_hello.py 
===================== test session starts =====================
platform linux -- Python 3.10.10, pytest-7.3.1, pluggy-0.13.1
rootdir: /home/ivanov_dmi/projects/pytest_fixture
collected 1 item                                                                                                                                                                                                                                                                                                   

test_hello.py Hello
world
.

```

**Reply by amercader** [reply]
No @NeroMorto , we ended up rewriting the tests to not rely on setup methods. Your `autouse=True` approach looks like a good workaround, but I seems like better to not rely on it long term.

**Comment by RonnyPfannschmidt** [comment]
I suspect this is related to turning xunit setup/teardown into autouse fixture

**Reply by NeroMorto** [reply]
@RonnyPfannschmidt Hi! I'm sorry for late answer.
In the end is there any way for current (latest) `pytest` version to execute fixture before `setup_method` (without `autouse=True` option)? 
Just like it was before `pytest 7` release.

**Reply by RonnyPfannschmidt** [reply]
Nope
